discovery/scaleway/instance.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	instanceLabelPrefix = metaLabelPrefix + "instance_"
+
+	instanceIDLabel         = instanceLabelPrefix + "id"
+	instanceIPIsPublicLabel = instanceLabelPrefix + "ip_is_public"
+	instanceIPVersionLabel  = instanceLabelPrefix + "ip_version"
+	instanceImageNameLabel  = instanceLabelPrefix + "image_name"
+	instanceNameLabel       = instanceLabelPrefix + "name"
+	instanceStateLabel      = instanceLabelPrefix + "status"
+	instanceTagsLabel       = instanceLabelPrefix + "tags"
+	instanceTypeLabel       = instanceLabelPrefix + "type"
+	instanceZoneLabel       = instanceLabelPrefix + "zone"
+)
+
+type instanceDiscovery struct {
+	*refresh.Discovery
+	Client    *scw.Client
+	Port      int
+	Zone      string
+	Project   string
+	AccessKey string
+	SecretKey string
+	Name      string
+	Tags      []string
+}
+
+func newInstanceDiscovery(conf *SDConfig) (*instanceDiscovery, error) {
+	d := &instanceDiscovery{
+		Port:      conf.Port,
```

Comment:
```suggestion
		port:      conf.Port,
```

I don't think we need to export these fields?